v1/feed.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+
+
+var P = require('bluebird');
+var HyperSwitch = require('hyperswitch');
+var uuid = require('cassandra-uuid').TimeUuid;
+
+var URI = HyperSwitch.URI;
+var HTTPError = HyperSwitch.HTTPError;
+
+var spec = HyperSwitch.utils.loadSpec(__dirname + '/feed.yaml');
+
+
+var DEFAULT_TTL = 3600;
+
+var FEED_URIS = {
+    tfa: { uri: ['v1', 'page', 'featured'], date: true },
+    mostread: { uri: ['v1', 'page', 'most-read'], date: true },
+    random: { uri: ['v1', 'page', 'random', 'title'], date: false },
+    image: { uri: ['v1', 'media', 'image', 'featured'], date: true },
+    news: { uri: ['v1', 'page', 'news'], date: false }
+};
+
+
+function Feed(options) {
+
+    this.options = options;
+
+}
+
+
+Feed.prototype._makeFeedRequests = function(parts, hyper, rp, dateArr) {
+
+    var self = this;
+    var props = {};
+
+    parts.forEach(function(part) {
+        var def = FEED_URIS[part];
+        var uriArray = [self.options.host, rp.domain].concat(def.uri);
+        if (def.date) {
+            Array.prototype.push.apply(uriArray, dateArr);
+        }
+        props[part] = hyper.get({
+            uri: uriArray.join('/'),
+            query: { aggregated: true }
+        });
+    });
+
+    return P.props(props);
+
+};
+
+
+Feed.prototype.aggregated = function(hyper, req) {
+
+    var self = this;
+    var rp = req.params;
+    var date;
+    var dateArr;
+
+    // key the records on the date in the format YYYY-MM-DD
+    try {
+        date = new Date(Date.UTC(rp.yyyy, rp.mm - 1, rp.dd));
+        date = date.toISOString().split('T').shift();
+        dateArr = date.split('-');
+    } catch (err) {
+        return P.reject(new HTTPError({
+            status: 400,
+            body: {
+                type: 'bad_request',
+                description: 'wrong date format specified'
+            }
+        }));
+    }
+
+    // check if we have a record in Cassandra already
+    return hyper.get({
+        uri: new URI([rp.domain, 'sys', 'key_value', 'feed.aggregated', date])
+    }).then(function(res) {
+        // we've got a cache hit, so we just need to request
+        // the random component and return the bundle
+        if (!res.body) {
+            return P.reject(new HTTPError({
```

Comment:
Happened to me during testing. I agree that this can happen only in an extreme circumstance, but it doesn't hurt to have it here.